### PR TITLE
[hhh] Add devicelab try builder

### PR DESCRIPTION
**Pull request by CaseyHillers**
This will allow the Dart team to experiment with hot_mode_dev_cycle_linux__benchmark on Android. They can verify their pipelines on their side before they use the existing prod builders.

### CODE REVIEW

**Comment by CaseyHillers** [comment]
\cc @whesse

File: config/devicelab_staging_config.star [hunk 1]
```diff
@@ -325,4 +325,46 @@ def devicelab_staging_prod_config():
...
+                },
+            ],
+            "task_name": "hot_mode_dev_cycle_linux__benchmark",
+            "upload_metrics": False,
+            "use_cas": True,
+            "attached_device": True,
+        },
+        os = "Android",
+        dimensions = {"device_os": "N"},
+        # TODO(keyonghan): adjust the timeout when devicelab linux tasks are stable:
```
Comment (godofredoc) [inline config/devicelab_staging_config.star:360]: Seems like this TODO is outdated

Comment (godofredoc) [reply config/devicelab_staging_config.star:360]: We should probably remove it, the associated bug is closed.

File: config/lib/repos.star [hunk 2]
```diff
@@ -16,6 +16,8 @@ repos = struct(
         "https://chromium.googlesource.com/external/github.com/flutter/engine",
     PACKAGES =
         "https://chromium.googlesource.com/external/github.com/flutter/packages",
+    DART_HHH =
```
Comment (godofredoc) [inline config/lib/repos.star:19]: How are we planning to trigger the build?

Comment (CaseyHillers) [reply config/lib/repos.star:19]: Dart will be triggering these builds when they run their HHH builds through buildbucket. This repo logic is part of the try_builder boiler plate, and will not be used by these try builds.

File: config/devicelab_staging_config.star [hunk 3]
```diff
@@ -357,13 +357,9 @@ def devicelab_staging_prod_config():
         },
         os = "Android",
         dimensions = {"device_os": "N"},
-        # TODO(keyonghan): adjust the timeout when devicelab linux tasks are stable:
-        # https://github.com/flutter/flutter/issues/72383.
-        expiration_timeout = timeout.XL,
         execution_timeout = timeout.MEDIUM,
         caches = LINUX_DEFAULT_CACHES,
         category = "Linux_android",
-        pool = "luci.flutter.try",
```
Comment (godofredoc) [inline config/devicelab_staging_config.star]: pool = 'luci.flutter.staging'?

**Review by godofredoc — APPROVED** [review]
LGTM after adding back the pool property and updating it to point to staging.

**Pull request merged**